entrypoint.sh

```diff
@@ -21,109 +23,47 @@ fi
-  black_error="false"
-  reviewdog_error="false"
-elif [[ "${black_exit_val}" -eq "1" && "${reviewdog_exit_val}" -eq "0" ]]; then
-  black_error="true"
-  reviewdog_error="false"
-elif [[ "${black_exit_val}" -eq "1" && "${reviewdog_exit_val}" -eq "1" ]]; then
-  black_error="true"
-  reviewdog_error="true"
-elif [[ "${black_exit_val}" -eq "0" && "${reviewdog_exit_val}" -eq "1" ]]; then
-  black_error="false"
-  reviewdog_error="true"
-else
-  if [[ "${black_exit_val}" -ne "0" && "${black_exit_val}" -ne "1" && \
-    "${reviewdog_exit_val}" -ne "0" && "${reviewdog_exit_val}" -ne "1" ]]; then
-    echo "[action-black] ERROR: Something went wrong while trying to run the black" \
-      "formatter while annotating the changes using reviewdog (black error code:" \
-      "${black_exit_val}, reviewdog error code: ${reviewdog_exit_val})."
-    exit 1
-  elif [[ "${black_exit_val}" -ne "0" && "${black_exit_val}" -ne "1" ]]; then
-    echo "[action-black] ERROR: Something went wrong while trying to run the black" \
-      "formatter (error code: ${black_exit_val})."
-    exit 1
-  else
-    echo "[action-black] ERROR: Something went wrong while trying to run the" \
-      "reviewdog error annotator (error code: ${reviewdog_exit_val})."
-    exit 1
-  fi
-fi
+  echo "[action-black] Checking python code with the black formatter and reviewdog..."
+  black_check_output="$(black --check ${black_args[*]} 2>&1)" ||
+    black_exit_val="$?"
 
-# Also format code if this is requested
-# NOTE: Useful for writing back changes or creating a pull request.
-black_format_exit_val="0"
-if [[ "${INPUT_FORMAT,,}" = 'true' && "${black_error}" = 'true' ]]; then
-  echo "[action-black] Formatting python code using the black formatter..."
-  black ${black_args[@]} || black_format_exit_val="$?"
+  # Intput black formatter output to reviewdog
+  echo "${black_check_output}" | reviewdog -f="black" \
+    -name="${INPUT_TOOL_NAME}" \
+    -reporter="${INPUT_REPORTER}" \
+    -filter-mode="${INPUT_FILTER_MODE}" \
+    -fail-on-error="${INPUT_FAIL_ON_ERROR,,}" \
+    -level="${INPUT_LEVEL}" \
+    ${INPUT_REVIEWDOG_FLAGS} || reviewdog_exit_val="$?"
+fi
 
-  # Check whether black formatting was succesfull
-  if [[ "${black_format_exit_val}" -eq "0" ]]; then
-    echo "::set-output name=is_formatted::true"
-  elif [[ "${black_format_exit_val}" -eq "1" ]]; then
-    black_error="true"
-    echo "::set-output name=is_formatted::false"
-  else
-    echo "[action-black] ERROR: Something went wrong while trying to run the" \
-      "black formatter (error code: ${black_format_exit_val})."
-    exit 1
-  fi
-elif [[ "${INPUT_FORMAT,,}" = 'true' && "${black_error}" = 'false' ]]; then
-  echo "[action-black] Formatting not needed."
-  echo "::set-output name=is_formatted::false"
-else
-  echo "::set-output name=is_formatted::false"
+# Catch black syntax error in order to make sure it is passed to reviewdog
+if [[ "${black_exit_val}" -eq "123" ]]; then
```

Comment:
I think here catching the 123 error makes sense since we know this is caused by a python syntax error and the `black` reviewdog errorformat is able to catch this error.

Reply:
But the script compares the exist code only with `1` at the last of the script, so can we remove this logic or change the last logic to check whether it's non-zero or not instead of comparing it to 1?

Reply:
That's true I updated it in the last commit.